django_cloud_deploy/integration_tests/tool_requirements_test.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Integration tests for tool_requirements.py."""
+import os
+
+from django_cloud_deploy import tool_requirements
+from django_cloud_deploy.cli import io
+from absl.testing import absltest
+
+
+class ToolRequirementsNegativeTest(absltest.TestCase):
+    """This will test all negative cases (not having the requirement).
+       It will remove any requirements in the setup.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Modify the path to exclude the necesary requirments."""
+        os.environ['PATH'] = ''
+
+    def setUp(self):
+        self.test_console = io.TestIO()
+
+    def requirement_missing(self, requirement):
+        with self.assertRaises(
+                tool_requirements.MissingRequirementsError) as cm:
+            tool_requirements.check_and_handle_requirements(self.test_console,
+                                                            [requirement])
+
+        exception = cm.exception
+        self.assertEqual(requirement.NAME,
+                         exception.missing_requirements[0].name)
+
+    def test_gcloud_missing(self):
+        gcloud = tool_requirements.Gcloud
+        self.requirement_missing(gcloud)
+
+    def test_kubectl_missing(self):
+        kubectl = tool_requirements.Kubectl
+        self.requirement_missing(kubectl)
+
+    def test_cloud_sql_proxy_missing(self):
+        cloud_sql_proxy = tool_requirements.CloudSqlProxy
+        self.test_console.answers.append('n')
+        self.requirement_missing(cloud_sql_proxy)
+
+    def test_docker_missing(self):
+        docker = tool_requirements.Docker
+        self.requirement_missing(docker)
+
+
+class ToolRequirementsPositiveTest(absltest.TestCase):
```

Comment:
Looks like this is unused. Maybe we can add it back after the test case is fully implemented.

Reply:
ack